### PR TITLE
APPS-1166 Fix CWL applet/job reuse

### PR DESCRIPTION
**Pull request by YuxinShi0423**

### CODE REVIEW

**Comment by YuxinShi0423** [comment]
TODO: need to fix the JSON lookup: the id of the target task might appear multiple times. Need to use full id instead.

File: compiler/src/main/scala/dx/translator/cwl/ProcessTranslator.scala [hunk 1]
```diff
@@ -80,10 +80,30 @@ import spray.json._
 import java.nio.file.{Path, Paths}
 import scala.util.Try
 
-case class CwlSourceCode(source: Path) extends SourceCode {
+case class CwlSourceCode(source: Path, target: Vector[String] = Vector.empty) extends SourceCode {
```
Comment (commandlinegirl) [inline compiler/src/main/scala/dx/translator/cwl/ProcessTranslator.scala:89]: If you could add a help string with a short description on the class (what the inputs are)?

**Review by commandlinegirl — DISMISSED** [review]
LGTM, with a minor request.

File: compiler/src/main/scala/dx/translator/cwl/ProcessTranslator.scala [hunk 2]
```diff
@@ -80,10 +80,30 @@ import spray.json._
...
   override val language: String = "cwl"
   override def toString: String = FileUtils.readFileContent(source)
-  def getDocContents: String = toString
+  override def getDocContents: String = {
```
Comment (Gvaihir) [inline compiler/src/main/scala/dx/translator/cwl/ProcessTranslator.scala:92]: Perfect - exactly where I left the placeholder in 994:)

File: compiler/src/test/scala/dx/translator/cwl/ProcessTranslatorTest.scala [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    standAloneProcessDoc1("count-lines1-wf_outputs") should not equal standAloneProcessDoc2(
+        "count-lines1-wf_outputs"
+    )
+    standAloneProcessDoc1("count-lines1-wf") should not equal standAloneProcessDoc2(
+        "count-lines1-wf"
+    )
+
+  }
+
+  "ProcessTranslator" should "render different cwl code for changed common/outputs" in {
```
Comment (Gvaihir) [inline compiler/src/test/scala/dx/translator/cwl/ProcessTranslatorTest.scala]: Just as a convention: class name to be tested is written only for the first test in the series of tests. All subsequent tests are named just `it`.

**Review by Gvaihir — COMMENTED** [review]
Good test coverage!

File: compiler/src/test/scala/dx/translator/cwl/ProcessTranslatorTest.scala (hunk 3 above)
Comment (YuxinShi0423) [reply compiler/src/test/scala/dx/translator/cwl/ProcessTranslatorTest.scala]: Thanks! I didn't notice that when copypasting. Changed as you suggested.

File: compiler/src/main/scala/dx/translator/cwl/ProcessTranslator.scala (hunk 1 above)
Comment (YuxinShi0423) [reply compiler/src/main/scala/dx/translator/cwl/ProcessTranslator.scala:89]: Added. Thanks!

Comment (commandlinegirl) [reply compiler/src/main/scala/dx/translator/cwl/ProcessTranslator.scala:89]: Thank you!

**Review by commandlinegirl — DISMISSED** [review]
LGTM!

**Review by commandlinegirl — DISMISSED** [review]

File: build.sbt [hunk 4]
```diff
@@ -132,7 +132,7 @@ val githubDxCompilerResolver = Resolver.githubPackages("dnanexus", "dxCompiler")
 
 lazy val dependencies =
   new {
-    val dxCommonVersion = "0.11.2"
+    val dxCommonVersion = "0.11.3-SNAPSHOT"
```
Comment (YuxinShi0423) [inline build.sbt:135]: Use https://github.com/dnanexus/dxScala/pull/108 which fixed JSON checksum calculation.
@commandlinegirl @Gvaihir

**Review by commandlinegirl — DISMISSED** [review]

**Review by commandlinegirl — DISMISSED** [review]

**Pull request merged**